modules/apps/portal/portal-verify-extender/src/main/java/com/liferay/portal/verify/extender/internal/osgi/commands/VerifyProcessTrackerOSGiCommands.java

```diff
@@ -179,15 +177,29 @@ public VerifyProcess addingService(
 					VerifyProcess verifyProcess = _bundleContext.getService(
 						serviceReference);
 
-					if (upgrading ||
-						_isInitialDeployment(serviceReference, verifyProcess)) {
+					String verifyProcessName = String.valueOf(
+						serviceReference.getProperty("verify.process.name"));
+
+					Release release = _releaseLocalService.fetchRelease(
+						verifyProcessName);
```

Comment:
@achaparro this line looks very much wrong. I know it is not from this pull, it was added way back by https://github.com/liferay/liferay-portal/commit/f8d59c68 . But since this piece of logic is touched, we should fix it.
It should not just assume verifyProcessName is a bundle's symbolic name. verifyProcessName is from "verify.process.name" service property which technically can be anything.
We should always get the bundle's symbolic name from the VerifyProcess instance's source bundle.

There is a bit other performance concern here, but not serious, please fix this bad logic first. I will continue the review afterward.

Reply:
Hi @shuyangzhou,

I realized about that when I was implementing this but I kept as it was to do not mix two different issues. Fixing this requires, apart from the logic changes, an upgrade process to update the servletContextName for some verifiers such as FacebookConnectCompanySettingsVerifyProcess whose verify.process.name does not match its bundle symbolic name.

Also, we will need to deal with the use case of having more than one verify process in a module, how we mark  the Release table the verify process that failed or is pending if the column verified is just a boolean? should we restrict the verify processes to one per module? (I think this makes sense since no one created more than one in all of these years)

Reply:
Ok, if you want to take care it separately, that's fine.
Single verifyprocess per module should be good. We need to add either SF level or code level protection logic.
I will do a round of performance review after the bundle symbolic name issue is fixed.
For now, as long as this does not cause noticeable startup performance issue, it is good to go.

@111Anna please run a startup performance check for this pull.

Reply:
Thank @shuyangzhou, I have created the following ticket: https://issues.liferay.com/browse/LPS-173444 and I will work on it once this pull is committed.

Cheers!